### ISSUE TITLE
logger instead of print in lr_scheduler.py

### ISSUE THREAD

**Issue by AngledLuffa** [pytorch/pytorch#100847]
### 🚀 The feature, motivation and pitch

In lr_scheduler.py, there are a couple calls to print().  For example, `ReduceLROnPlateau._reduce_lr` calls `print`.  If it called a logger instead, the logs could be turned off, or at least would show up in python logging streams as part of stderr.

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

**Comment by janeyx99** [comment]
We would accept a PR replacing the print statements in lr scheduler with a logger!

**Comment by janeyx99** [comment]
On second thought, I do agree with @albanD that pytorch is not a service for which logging (nor printing) would make the most sense. Removing the verbose option seems like a logical next step.

**Comment by AngledLuffa** [comment]
That's fair, but removing the verbose option would change the API for
everyone who currently has it verbose.  Would it be better to simply make
the verbose option do nothing?

**Comment by AngledLuffa** [comment]
Doing a quick scan, it appears each scheduler has a `verbose` argument which is used to call `print_lr`, which does some `print` statements rather than log statements (the point of my original question).  In addition, there is a `print` statement in `ReduceLROnPlateau`'s `_reduce_lr` statement.

Is your suggestion to remove all of those `print` statements and warn that `verbose` is deprecated?  Is that my department still?  I was originally only concerned with the `_reduce_lr`'s `print`, although simply creating the scheduler with `verbose=False` on our end would also make the clutter at the bottom of our outputs go away.

Also, as a random aside, `ReduceLROnPlateau` is not a subclass and seems to get away with being a scheduler via duck typing.  Is that interpretation correct?

**Comment by janeyx99** [comment]
Yes, #101099 seems to do a pretty good job listing out all the print statements for lr schedulers. My suggestion _is_ to remove all the print statements + warn that `verbose` is deprecated. However, nothing prevents a smaller step to that bigger suggestion with just the `_reduce_lr`'s print statement going away. That said, I do not envision propagating this change to the other LR schedulers to be significantly harder.

And about the aside: I am not an expert on LR Schedulers, and your interpretation is correct. I'm not sure why it's not a subclass (maybe it's a mistake?) but the origins reach all the way back to the beginning: https://github.com/pytorch/pytorch/commit/630af4d7d8812efa43a01d938f0fc53c47802b0b
